### PR TITLE
Spotlight search proposed changes

### PR DESCRIPTION
**Pull request by nickoferrall**
Proposed changes to the search query following some merge conflict fun.

### CODE REVIEW

File: packages/client/hooks/useGroupMatrix.ts
Comment (nickoferrall) [inline packages/client/hooks/useGroupMatrix.ts:32]: When we update the search query, `queryRef` is briefly undefined which means we don't render the results. This means that the `resultsRef` is undefined so the groupMatrix is set to `[]`. To prevent this, I've moved `initMatrix` to the `useEffect` and only update state if `columnsCount` isn't null.

File: packages/client/hooks/useGroupMatrix.ts
Comment (nickoferrall) [inline packages/client/hooks/useGroupMatrix.ts:67]: Previously, if a results group was grouped/ungrouped it would be added or removed from the matix. With the search query, many groups could be added or removed, so I've updated the logic.

**Review by nickoferrall — COMMENTED** [review]
Hey @tianrunhe, I've made these changes in a separate PR so you can choose whether or not to accept them. 

I think one issue that'll need to be resolved before the search query can be merged is the modal flashes after each keystroke. We discussed this before and I think it's become more noticeable after the component order was changed. I'll continue to look for a solution now, but feel free to merge this PR into yours if you're happy with it.

File: packages/client/hooks/useQueryLoaderNow.tsx
Comment (nickoferrall) [inline packages/client/hooks/useQueryLoaderNow.tsx:27]: This logic prevents the results from suspending on each keystroke, which causes the modal to flash. It could probably be cleaned up a little - I don't like the name `refreshQuery` for example when we have `refresh` lower down in the file

**Comment by nickoferrall** [comment]
The last two commits fixed the modal flashing on each keystroke 🎉 

One final change could be preventing the height of the modal from changing when the results change. One way I was thinking about implementing that was passing the search query into [useResultsHeight](https://github.com/ParabolInc/parabol/blob/541b938b1fd92fec20a00b38b0a6c11b050fd279/packages/client/hooks/useResultsHeight.ts) and preventing the height from updating if `spotlightSearchQuery !== ''`, but it's up to you. 

I'll pass the baton back to you now!

**Review by tianrunhe — APPROVED** [review]

**Pull request merged**
